components/eamxx/src/share/util/scream_vertical_interpolation.hpp

```diff
@@ -128,16 +119,62 @@ void perform_vertical_interpolation_impl_1d(
   const MemberType& team,
   const LIV<T,P>& vert_interp);
 
+/* ---------------------------------------------------------------------- 
+ * Versions where x_tgt is a 2-D view
+ * ---------------------------------------------------------------------- */
 template<typename T, int P, int N> 
-void perform_vertical_interpolation_impl_Nd(
-  const view_Nd<const Pack<T,P>,2>& x_src,
+void perform_checks(
+  const view_2d<const Pack<T,P>>&   x_src,
+  const view_2d<const Pack<T,P>>&   x_tgt,
+  const view_Nd<const Pack<T,P>,N>& input,
+  const view_Nd<      Pack<T,P>,N>& output,
+  const int                         nlevs_src,
+  const int                         nlevs_tgt);
+
+template<typename T, int P> 
+void apply_interpolation(
+  const                        int  num_levs,
+  const                          T  mask_val,
+  const                   LIV<T,P>& vert_interp,
+  const view_2d<const Pack<T,P>>&   x_src,
+  const view_2d<const Pack<T,P>>&   x_tgt,
+  const view_Nd<const Pack<T,P>,2>& input,
```

Comment:
Any reason not to use `view_2d<T>` instead of `view_Nd<T,2>`?

Reply:
No reason then just a bunch of copy/pastes and changing a single number.  I can change this and all others to `view_2d`, `view_3d`, etc. if that is desired.

Reply:
It's not really a big deal, but since you are already using view_2d, I think it looks cleaner. At least to me. No pressure though, do what you like best.